MaxText/layers/decoders.py

```diff
@@ -587,11 +587,11 @@ def _apply_output_head(self, y, deterministic, model_mode):
   @nn.compact
   def __call__(
       self,
+      shared_embedding: nn.Module | nnx.Module,
```

Comment:
Why do we need to pass this `shared_embedding` around now?

Reply:
In principle we can, at some point @bvandermoon and I thought it was better to only have it in the constructor but maybe we can have both.